geopandas/array.py

```diff
@@ -466,12 +466,44 @@ def touches(self, other):
     def within(self, other):
         return self._binary_method("within", self, other)
 
+    def geom_equals_exact(self, other, tolerance):
+        return self._binary_method("equals_exact", self, other, tolerance=tolerance)
+
+    def geom_almost_equals(self, other, decimal):
+        return self.geom_equals_exact(other, 0.5 * 10 ** (-decimal))
+        # return _binary_predicate("almost_equals", self, other, decimal=decimal)
+
+    # compatibility with pandas 1.1
+    def equals(self, other):
+        if compat.PANDAS_GE_11:
+            warnings.warn(
+                "GeometryArray.equals() is now GeometryArray.geom_equals(). "
+                "For geometry equality check use GeometryArray.geom_equals() instead."
+                "GeometryArray.equals() returns ExtensionArray.equals().",
```

Comment:
Do you think this warning is needed? I would not expect that many people would use `GeometryArray` directly, they probably rather using GeoSeries. And also, you already made a hard break in the behaviour (and I think that is fine), but so this warning is only indicating that it already broke, not that it will break in the future?

Reply:
If we're fine with hard break without a warning (even ex-post) then it is not needed. It is true that people likely do not use it on array level anyway.

Reply:
Yeah, I think simply changing this is fine